src/sqlancer/citus/README.md

```diff
@@ -0,0 +1,141 @@
+# SQLancer for Citus (PostgreSQL extension)
+
+SQLancer (Synthesized Query Lancer) is a tool to automatically test Database Management Systems (DBMS) in order to find logic bugs in their implementation. More information about the tool can be found in the [SQLancer README](https://github.com/sqlancer/sqlancer).
+
+The Citus implementation of SQLancer supports the Ternary Logic Query Partitioning (TLP) test oracle.
+
+# Setting up
+
+These instructions are taken from the SQLancer README and tailored specifically for the Citus implementation.
+
+Requirements:
+* Java 8 or above
+* [Maven](https://maven.apache.org/) (`sudo apt install maven` on Ubuntu)
+* PostgreSQL & Citus - The steps required to build Citus from source can be found [here](https://github.com/citusdata/citus/blob/master/CONTRIBUTING.md).
+Optional Requirements:
+* [pgenv](https://github.com/thanodnl/pgenv) (for easier management of PostgreSQL versions)
+* [citus_dev](https://github.com/citusdata/tools/tree/develop/citus_dev) (for easier configuration of Citus environment)
+
+The following commands clone SQLancer and create a JAR file:
+
+```
+git clone https://github.com/sqlancer/sqlancer
+cd sqlancer
+mvn package -DskipTests
+```
+
+# Using SQLancer
+
+The following commands run the Citus implementation of SQLancer using Ternary Logic Query Partitioning (TLP):
+
+```
+cd target
+java -jar SQLancer-0.0.1-SNAPSHOT.jar --num-threads 4 citus --oracle QUERY_PARTITIONING
+```
+
+How to configure the run and how to interpret the output is explained [here](https://github.com/sqlancer/sqlancer#using-sqlancer).
+
+Here is a list of main and Citus implementation-specific options that can be passed in as flags using the following format:
```

Comment:
The option list will likely be outdated soon. Please remove this, or restrict it to the Citus-specific options, if you expect them to not change, or if you want to keep this part up-to-date.